src/scheduler/fifo.cpp

```diff
@@ -1392,7 +1392,10 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int has_runjob_hook, s
 	if (rc) {
 		char buf[MAX_LOG_SIZE];
 		set_schd_error_codes(err, NOT_RUN, RUN_FAILURE);
-		errbuf = pbs_geterrmsg(get_svr_inst_fd(pbs_sd, rjob->job->svr_inst_id));
+
+		if (svrinstfd_map.find(rjob->job->svr_inst_id) == svrinstfd_map.end())
```

Comment:
Doesn't get_svr_inst_fd() iterate over like 5 things?  I don't thinkt his umap is going to help much and adds complexity that I don't see as needed.  If you want it, please test it to see if it really helps.

Reply:
as we discussed offline, it'll avoid string parsing that parse_servername() does, and the loop around servers, which can theoretically be up to 100, for each job. Using the map doesn't really add much complexity, but I will simplify this to move get_svr_inst_fd inside the scheduler, or create a scheduler wrapper for it so that the map is limited to that function, as Arun suggested.